go/ql/src/experimental/CWE-639/urlCheck.ql

```diff
@@ -0,0 +1,95 @@
+/**
+ * @name Incorrect check on url
+ * @description If a CORS policy is configured to accept an origin value obtained from the request data, it can lead to a policy bypass.
+ * @kind path-problem
+ * @problem.severity warning
+ * @id go/cors-bypass
+ * @tags security
+ *       experimental
+ *       external/cwe/cwe-942
+ *       external/cwe/cwe-346
+ */
+
+import go
+
+bindingset[s]
+private predicate mayBeCors(string s) { s.toLowerCase().matches(["%origin%", "%cors%"]) }
+
+/**
+ * An argument to a Gorilla's OriginValidator Function taken as a source
+ */
+class GorillaOriginFuncSource extends RemoteFlowSource::Range {
```

Comment:
```suggestion
class GorillaOriginFuncSource extends DataFlow::Node {
```
I don't think you actually want this to extend `RemoteFlowSource::Range`. Since this is a `.ql` file, it can't be imported by any other files, so the only effects are within this file. Looking at the uses of `RemoteFlowSource` in this file, the only effect is that `MaybeOrigin` can include nodes from `GorillaOriginFuncSource`. But as `MaybeOrigin` is only used in `node instanceof MaybeOrigin or node instanceof GorillaOriginFuncSource`, this makes no difference.